### ISSUE TITLE
Unable to alert on windows logon brute force attack(<same_user /> rule not working)

### ISSUE THREAD

**Issue by shahmilan34** [wazuh/wazuh#2524]
## Rule to detect brute force attack:
```xml
<group name="windows,">
        <rule id="100010" level="10" frequency="8" timeframe="240" >
                <if_matched_sid>20030</if_matched_sid>
                 <same_user />
                <description>Brute Force Attack ,Windows Multiple Failed login attemps</description>
                <group>win_authentication_failed,</group>
        </rule>
</group>
```
## Windows Agent config:
wazuh-agent v3.8.2
windows 8

## Wazuh-Manager config:
centos 7
wazuh-manager v3.8.2
Installation Method : rpm

Rule works if <same_user /> tag removed.

**Comment by cristgl** [comment]
Hello @shahmilan34,

You are right, these options like `<same_user />`, `<same_src_port />`, etc. are not matching with the events retrieved from the new Eventchannel. This is because the old decoder was filling variables as user, id, location, etc. But this decoder is not used anymore. I have opened an issue where I specify the new proposal for a dynamic option: https://github.com/wazuh/wazuh/issues/2531.

Thank you very much for reporting this.
Cristina

**Comment by cristgl** [comment]
Hello @shahmilan34, 

I am glad to tell you that this feature will be available at Wazuh v3.9.0. Now you can filter by `same_user` by using the new dynamic `same_field` option, developed at https://github.com/wazuh/wazuh/pull/2689. Also, we have already added it to the eventchannel ruleset at this PR: https://github.com/wazuh/wazuh-ruleset/pull/302.

With that said, your rule should look like the next one:
```
<group name="windows,">
        <rule id="100010" level="10" frequency="8" timeframe="240" >
                <if_matched_sid>20030</if_matched_sid>
                 <same_field>EventChannel.EventData.TargetUserSid<same_field/>
                <description>Brute Force Attack ,Windows Multiple Failed login attemps</description>
                <group>win_authentication_failed,</group>
        </rule>
</group>
```

Best regards,
Cristina

**Comment by chemamartinez** [comment]
Hi @shahmilan34,

Let me close the issue since PR #2689 has already been merged to the 3.9 branch. It will be available in the newly released version.

Don't hesitate to reopen it if you need any further assistance.

Best regards,
Chema.
